src/huggingface_hub/hf_api.py

```diff
@@ -529,17 +533,20 @@ def list_models(
                     >>> # List all models
                     >>> api.list_models()
 
+                    >>> # Get all valid search arguments
+                    >>> args = ModelSearchArguments()
+
                     >>> # List only the text classification models
                     >>> api.list_models(filter="text-classification")
+                    >>> # Using the `ModelFilter`
+                    >>> filt = ModelFilter(task="text-classification")
+                    >>> # With `ModelSearchArguments`
+                    >>> filt = ModelFilter(task=args.pipeline_tags.TextClassification)
+                    >>> api.list_models(filter=filt)
 
-                    >>> # List only the russian models compatible with pytorch
-                    >>> api.list_models(filter=("ru", "pytorch"))
-
-                    >>> # List only the models trained on the "common_voice" dataset
-                    >>> api.list_models(filter="dataset:common_voice")
+                    >>> # Using ModelFilter and SearchArguments to find text classification in both PyTorch and TensorFlow
+                    >>> filt = ModelFilter(task=args.pipeline_tags.TextClassification, framework=[args.library.PyTorch, args.library.TensorFlow])
 
-                    >>> # List only the models from the AllenNLP library
-                    >>> api.list_models(filter="allennlp")
```

Comment:
Could we keep one example with a filter for a 3rd party library to have as a pointer?